plugins/com.yakindu.solidity.ide/src/com/yakindu/solidity/ide/internal/SolidityIdeCodeActionService.xtend

```diff
@@ -1,48 +1,64 @@
 package com.yakindu.solidity.ide.internal
 
 import com.google.inject.Inject
-import com.google.inject.name.Named
-import com.yakindu.solidity.SolidityVersion
-import com.yakindu.solidity.validation.IssueCodes
-import java.util.List
 import org.eclipse.emf.common.util.URI
 import org.eclipse.lsp4j.CodeAction
-import org.eclipse.lsp4j.CodeActionParams
-import org.eclipse.lsp4j.Command
 import org.eclipse.lsp4j.TextEdit
 import org.eclipse.lsp4j.WorkspaceEdit
 import org.eclipse.lsp4j.jsonrpc.messages.Either
-import org.eclipse.xtext.ide.server.Document
-import org.eclipse.xtext.ide.server.codeActions.ICodeActionService
-import org.eclipse.xtext.resource.XtextResource
-import org.eclipse.xtext.util.CancelIndicator
+import org.eclipse.xtext.ide.server.codeActions.ICodeActionService2
+import com.yakindu.solidity.SolidityVersion
+import com.google.inject.name.Named
 
-class SolidityIdeCodeActionService implements ICodeActionService {
+import static com.yakindu.solidity.validation.IssueCodes.*
+
+class SolidityIdeCodeActionService implements ICodeActionService2 {
 
 	@Inject @Named(SolidityVersion.SOLIDITY_VERSION) String solcVersion
 
-	override List<Either<Command, CodeAction>> getCodeActions(Document document, XtextResource resource,
-		CodeActionParams params, CancelIndicator indicator) {
+	protected def addTextEdit(WorkspaceEdit edit, URI uri, TextEdit... textEdit) {
+		edit.changes.put(uri.toString, textEdit)
+	}
+	
+	override getCodeActions(Options options) {
+		val resource = options.resource
+		val params = options.codeActionParams
 		val actions = newArrayList
 		for (d : params.context.diagnostics) {
-			if (d.code == IssueCodes.WARNING_SOLIDITY_VERSION_NOT_THE_DEFAULT) {
+			if (d.code.hasSolution) {
 				actions += Either.forRight(new CodeAction => [
-					title = '''Change to «solcVersion»'''
+					title = d.code.label
 					diagnostics = #[d]
 					edit = new WorkspaceEdit() => [
 						addTextEdit(resource.URI, new TextEdit => [
 							range = d.range
-							newText = solcVersion
+							newText = d.code.fix
 						])
 					]
 				])
 			}
 		}
 		return actions
 	}
-
-	protected def addTextEdit(WorkspaceEdit edit, URI uri, TextEdit... textEdit) {
-		edit.changes.put(uri.toString, textEdit)
+	
+	
+	def String getLabel(String issueCode) {
```

Comment:
should this be replaced by CodeActionProvider ?

Reply:
In the future yes, just realized that i accidentaly commited these files